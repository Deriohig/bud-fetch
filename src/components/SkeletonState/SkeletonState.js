import React from 'react';
import styled from "styled-components";
import PropTypes from 'prop-types';

const Skeleton = styled.p`
    position: relative;
    padding: 0;
    background: #e5e5e5;
    border: none;
    box-shadow: none;
    pointer-events: none;
    width: ${props => props.width ? props.width: '100%'};
    height: 16px;
    margin-bottom: 8px;
    @keyframes skeleton {
        0% {
          right: auto;
          left: 0;
          width: 0%;
          opacity: 0.3;
        }
        20% {
          right: auto;
          left: 0;
          width: 100%;
          opacity: 1;
        }
        28% {
          right: 0;
          left: auto;
          width: 100%;
        }
        51% {
          right: 0;
          left: auto;
          width: 0%;
        }
        58% {
          right: 0;
          left: auto;
          width: 0%;
        }
        82% {
          right: 0;
          left: auto;
          width: 100%;
        }
        83% {
          right: auto;
          left: 0;
          width: 100%;
        }
        96% {
          right: auto;
          left: 0;
          width: 0%;
        }
        100% {
          right: auto;
          left: 0;
          width: 0%;
          opacity: 0.3;
        }
      }
    &:after {
        position: absolute;
        top: 0;
        left: 0;
        width: 0%;
        height: 100%;
        background-color: #c6c6c6;
        animation: 3000ms ease-in-out skeleton infinite;
        content: '';
    }
`
const SkeletonState = ({width}) => {
  return (
       <Skeleton data-testid='SkeletonState' width={width}></Skeleton>
  );
};

SkeletonState.propTypes = {
  width: PropTypes.string
}

export default SkeletonState;
