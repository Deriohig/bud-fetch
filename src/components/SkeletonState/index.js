import Skeleton from './SkeletonState';

export default Skeleton;
