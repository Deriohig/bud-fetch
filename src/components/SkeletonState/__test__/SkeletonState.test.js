import React from 'react';
import ReactDOM from 'react-dom';
import SkeletonState from '../index';
import {render, cleanup} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer'

const props = {
    width: '30%'
}

afterEach(cleanup);

it("renders without crashing", ()=> {
    const div = document.createElement('div');
    ReactDOM.render(<SkeletonState {...props} />, div)
})

it("renders with a certain width", ()=> {
    const { getByTestId } = render(<SkeletonState {...props} />)
    const style = window.getComputedStyle(getByTestId('SkeletonState'));
    expect(style.width).toBe('30%');
})

it("matches snapshot", ()=> {
    const snapshot = renderer.create(<SkeletonState {...props} />).toJSON();
    expect(snapshot).toMatchSnapshot();
})
