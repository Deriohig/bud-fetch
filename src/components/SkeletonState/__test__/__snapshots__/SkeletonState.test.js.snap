// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<p
  className="sc-bdfBwQ knDrYM"
  data-testid="SkeletonState"
  width="30%"
/>
`;
